framework/include/base/MooseFunctor.h

```diff
@@ -406,44 +399,47 @@ class Functor : public FunctorBase
   /// (cached) value. If the boolean is false, then the value cannot be trusted
   mutable std::unordered_map<dof_id_type, std::vector<std::vector<std::pair<bool, T>>>>
       _side_qp_to_value;
+
+  template <typename>
+  friend class Functor;
 };
 
 template <typename T>
-typename Functor<T>::ValueType
-Functor<T>::operator()(const Elem * const & elem, const unsigned int state) const
+typename FunctorImpl<T>::ValueType
+FunctorImpl<T>::operator()(const Elem * const & elem, const unsigned int state) const
 {
   return evaluate(elem, state);
 }
```

Comment:
If you look at the quadrature point implementations, you will see that they support caching the evaluations, so just because you call `operator()` you may not get an `evaluate` call for those overloads. We could choose to do the same for this overload in the future if we wanted because we have the public facing `operator()` and then the private implementation method `evaluate`

Reply:
We've had users tell us that material property evaluations can be very expensive so they'd like to see a potential for caching

Reply:
Ahhh, I see, thanks.